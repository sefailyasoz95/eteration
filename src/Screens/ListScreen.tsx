import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import React, { useEffect, useState } from "react";
import { ActivityIndicator, StyleSheet, View } from "react-native";
import Button from "../Components/Button/Button";
import SimpsonsList from "../Components/Simpsons/SimpsonsList";
import { AppStackParamList } from "../StackParamLists/AppStackParamList";
import { getSimpsons } from "../Utils/Services/service";

type Props = {
	navigation: NativeStackNavigationProp<AppStackParamList, "List">;
};
const ListScreen = ({ navigation }: Props) => {
	const [screenState, setScreenState] = useState({
		dataSource: [],
		loading: true,
	});
	const handleData = async () => {
		setScreenState({ dataSource: [], loading: true });
		let result = await getSimpsons();
		setScreenState({ dataSource: result.data, loading: false });
	};

	useEffect(() => {
		// test çalıştırıldığında bu fonksiyonun async olup await'lenmediğinden dolayı kızıyor
		// açıkçası bilmiyorum nasıl giderilir bu uyarı.
		handleData();
	}, []);

	return (
		<View style={styles.container} testID='container'>
			{screenState.loading ? (
				<ActivityIndicator size={50} color='red' testID='indicator' />
			) : (
				<SimpsonsList
					dataSource={screenState.dataSource}
					navigation={navigation}
					refreshData={handleData}
					testID='simpson-flatlist'
				/>
			)}
			<Button
				text={`+`}
				testID={"+"}
				textStyle={styles.buttonText}
				buttonStyle={styles.buttonStyle}
				onPress={() => navigation.navigate("Create")}
			/>
		</View>
	);
};

const styles = StyleSheet.create({
	container: {
		flex: 1,
		alignItems: "center",
		justifyContent: "flex-start",
	},
	buttonText: {
		fontSize: 50,
	},
	buttonStyle: {
		position: "absolute",
		bottom: "3%",
		right: "5%",
	},
});

export default ListScreen;
