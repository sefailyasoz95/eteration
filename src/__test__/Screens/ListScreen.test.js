import React from "react";
import ListScreen from "../../Screens/ListScreen";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import { create } from "react-test-renderer";
configure({ adapter: new Adapter() });
const navigation = {
	navigate: jest.fn(),
};
describe("list screen", () => {
	it("should render screen correctly", async () => {
		const component = shallow(<ListScreen />);
		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it("it should simulate the plus button press and navigation to create screen", async () => {
		const component = create(<ListScreen navigation={navigation} />);
		const button = component.root.findByProps({ testID: "+" }).props;
		button.onPress();

		expect(navigation.navigate).toBeCalledWith("Create");
	});
});
