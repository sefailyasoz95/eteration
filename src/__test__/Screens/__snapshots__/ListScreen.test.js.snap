// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list screen should render screen correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
  testID="container"
>
  <ActivityIndicator
    animating={true}
    color="red"
    hidesWhenStopped={true}
    size={50}
    testID="indicator"
  />
  <Button
    buttonStyle={
      Object {
        "bottom": "3%",
        "position": "absolute",
        "right": "5%",
      }
    }
    onPress={[Function]}
    testID="+"
    text="+"
    textStyle={
      Object {
        "fontSize": 50,
      }
    }
  />
</View>
`;
